core/message_pool_test.go

```diff
@@ -571,3 +774,24 @@ func (p *storeBlockProvider) GetBlock(ctx context.Context, cid cid.Cid) (*types.
 	}
 	return &blk, nil
 }
+
+func setNonce(signer types.Signer, message *types.SignedMessage, nonce types.Uint64) *types.SignedMessage {
+	return resignMessage(signer, message, func(m *types.Message) {
+		m.Nonce = nonce
+	})
+}
+
+func resignMessage(signer types.Signer, message *types.SignedMessage, f func(*types.Message)) *types.SignedMessage {
+	var msg types.Message
+	msg = message.Message
+	f(&msg)
+	smg, err := signMessage(signer, msg)
+	if err != nil {
+		panic("Error signing message")
```

Comment:
nit: if we are going to panic here let's call this `mustResignMessage`

Reply:
done